### PR TITLE
frontend/ fix movies context and create auth and user context

### LINKED ISSUE #74: FRONTEND | Prepare Context for user and session

### PR DESCRIPTION
**Pull request by vieraboschkova**
closes #74

### CODE REVIEW

File: client/src/context/Auth/authReducer.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      };
+    case AUTH_ERROR:
+    case LOGOUT:
+      return {
+        ...state,
+        token: null,
+        isAuthenticated: false,
+        loading: false,
+        user: null,
+      };
```
Comment (aleksanderwalczuk) [inline client/src/context/Auth/authReducer.js]: looks the same to me 🤷

File: client/src/context/Auth/index.js [hunk 2]
```diff
@@ -1,9 +1,23 @@
...
-  return <div />;
-}
+const AuthContextProvider = ({ children }) => {
+  const [user, setUser] = useState(null);
+  const [isAuthenticated, setIsAuthenticated] = useState(false);
+  // const [isLoaded, setIsLoaded] = useState(false);
 
-export default AuthComponent;
+  return (
+    <div>
```
Comment (aleksanderwalczuk) [inline client/src/context/Auth/index.js]: Do we need that div?

File: client/src/context/Auth/index.js [hunk 3]
```diff
@@ -1,9 +1,23 @@
...
 
-export const AuthContext = createContext(['', () => {}]);
+export const AuthContext = createContext();
 
-function AuthComponent() {
-  return <div />;
-}
+const AuthContextProvider = ({ children }) => {
+  const [user, setUser] = useState(null);
+  const [isAuthenticated, setIsAuthenticated] = useState(false);
```
Comment (aleksanderwalczuk) [inline client/src/context/Auth/index.js]: why do you use useState in context?

File: client/src/context/Movies/index.js [hunk 4]
```diff
@@ -1,12 +1,27 @@
...
+    // console.log('from local storage: ', localMovies);
+    return localMovies ? JSON.parse(localMovies) : moviesMock;
+  });
+  useEffect(() => {
+    localStorage.setItem('movies', JSON.stringify(movies));
+  }, [movies]);
+  // const fetchAndSetMovies = () => {
+  //   console.log('fetching and setting movies');
+  //   // setMovies(moviesMock);
+  // };
```
Comment (aleksanderwalczuk) [inline client/src/context/Movies/index.js]: this is messy, please clean it up

**Review by paula0403 — APPROVED** [review]

**Pull request merged**